Bottom: 3cd70ab65f64093bd5d63f327933c2d0c1265253
Top:    ac89d9e57cb56c28aafce43703810b692da2775a
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-02-17 15:46:31 +0800

Refresh of fixup

---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index d11f6af..776bad6 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -418,7 +418,7 @@ static struct sk_buff *receive_small(struct net_device *dev,
 				     struct receive_queue *rq,
 				     void *buf, unsigned int len)
 {
-	struct sk_buff *skb = buf;
+	struct sk_buff *skb;
 	struct bpf_prog *xdp_prog;
 	unsigned int headroom = vi->hdr_len + NET_IP_ALIGN +
 		                virtnet_get_headroom(vi);
@@ -426,7 +426,7 @@ static struct sk_buff *receive_small(struct net_device *dev,
 		              SKB_DATA_ALIGN(sizeof(struct skb_shared_info));
 
 	len -= vi->hdr_len;
-
+#if 0
 	rcu_read_lock();
 	xdp_prog = rcu_dereference(rq->xdp_prog);
 	if (xdp_prog) {
@@ -462,6 +462,7 @@ static struct sk_buff *receive_small(struct net_device *dev,
 		}
 	}
 	rcu_read_unlock();
+#endif
 
 	skb = build_skb(buf, buflen);
 	if (!skb) {
